<launch>
  <param name="global_frame_id" value="/base_link" />	
<!--publish a point cloud to "cloud_pcd" topic-->
<!--output: cloud_pcd -->
<node pkg="point_cloud_mapping" type="pcd_to_msg_node" name="pcd_to_msg_node" output="screen" respawn="true"
      args="$(find cloud_tools)/data/teapot.pcd $(optenv period 5)" />


<!-- Box Filter, cuts out the box part of the pointcloud -->
<!--input: /multi_pcd_to_msg_node/cloud_pcd -->
<!--output: /shoulder_cloud_clipped -->
<node pkg="cloud_tools" type="box_filter_node" name="box_filter_node" output="screen">
  <remap from="~cloud_pcd" to="cloud_pcd" />
  <param name="output_cloud_topic" value="cloud_clipped" />
  <param name="box_min_x" value="0.0" />
  <param name="box_max_x" value="1.3" />
  <param name="box_min_y" value="-1.0" />
  <param name="box_max_y" value="1.0" />
  <param name="box_min_z" value="-0.5" />
  <param name="box_max_z" value="0.5" />
</node>


<!-- Table Object Detector (passive version) -->
<!--input: /shoulder_cloud_clipped -->
<!--output: 
    /table_with_objects -  
    /table_object_detector_passive_node/cloud_annotated - clusters
    /table_object_detector_passive_node/semantic_polygonal_map - table polygon
-->
<node pkg="cloud_tools" type="table_object_detector_passive_node" name="table_object_detector_passive_node" output="screen">
  <remap from="~cloud_pcd" to="box_filter_node/cloud_clipped" />
  <param name="table_delta_z" value="0.02" />
  <param name="output_table_topic" value="/table_with_objects" />
  <param name="sac_distance_threshold" value="0.015" />
  <param name="normal_eps_angle" value="5" />
  <param name="table_min_height" value="-1.0" />
  <param name="table_delta_z" value="0.01" />
</node>
</launch>
